src/common/buttons/BackButton.js

```diff
@@ -9,45 +9,60 @@ type Props = {|
   text: string,
   history: {
     goBack: () => void,
+    push: string => void,
+    location: {
+      pathname: string,
+    },
+    entries: Array<{
+      pathname: string,
+    }>,
   },
 |};
 
-const BackButton = (props: Props) => (
-  <div
-    data-cy="back-button"
-    className="back"
-    onClick={props.history.goBack}
-    onKeyUp={props.history.goBack}
-    role="button"
-    tabIndex="-1"
-  >
-    <div className="chevron">
-      <ChevronLeft size="small" customColor="#8291a6" />
-    </div>
-    <Typography type="secondary" variant="bold">
-      {props.text}
-    </Typography>
-    <style jsx>
-      {`
-        div.back {
-          position: absolute;
-          top: 18px;
-          left: 27px;
-          cursor: pointer;
-        }
-        div.chevron {
-          display: inline-block;
-          margin-right: 4px;
-        }
-        @media only screen and (max-width: 1181px) {
+const BackButton = (props: Props) => {
+  const { location, entries } = props.history;
+  const firstEntry = entries.length === 1 && entries[0];
+  const goTo = () =>
```

Comment:
Well, I'd say `navigate` instead of `goTo`, but also it seems it's doing something else than what the name says, no?
Is it basically a more intelligent version of `history.goBack`? If so I think it's fine to name it `goBack`.